peakina/readers/excel.py

```diff
@@ -1,70 +1,308 @@
+        to_iter = range(preview_nrows + 1)
+    else:
+        to_iter = range(wb.sheet_by_name(sh_name).nrows)
+
+    for rx in to_iter:
+        yield wb.sheet_by_name(sh_name).row(rx)
+
+
+def _new_xls_rows_iterator(
+    wb: Union[openpyxl.workbook.Workbook, xlrd.book.Book],
+    sh_name: str,
+    preview_nrows: Optional[int],
+    preview_offset: Optional[int],
+) -> Generator[Any, Any, Any]:
+    """
+    Depending on paginations inputs (preview_rows, preview_offset), we want to
+    get an iterator object to loop on target rows, here we're returning an iterator
+    from the iter_rows built-in function from openpyxl
+
+    """
+
+    # +1 are here because this is 1-based indexing
+    if preview_nrows is not None and preview_offset is not None:
+        max_row = preview_offset + 1 + preview_nrows
+    elif preview_nrows is not None and preview_offset is None:
+        max_row = preview_nrows + 1
+    else:
+        max_row = None
+
+    if preview_offset:
+        min_row = preview_offset + 1
+    else:
+        min_row = None
+
+    # Then we return the generator
+    yield wb[sh_name].iter_rows(
+        min_row=min_row,
+        max_row=max_row,
+        values_only=True,
+    )
+
+
+def _get_rows_iterator(
+    wb: Union[openpyxl.workbook.Workbook, xlrd.book.Book],
+    sheet_name: str,
+    preview_nrows: Optional[int],
+    preview_offset: Optional[int],
+) -> Generator[Any, Any, Any]:
+    """
+    Depending on the excel type either it's the new format or the old one,
+    this method will return an iterator to read on its rows
+    """
+
+    if isinstance(wb, xlrd.book.Book):
+        return _old_xls_rows_iterator(wb, sheet_name, preview_nrows, preview_offset)
+
+    return _new_xls_rows_iterator(wb, sheet_name, preview_nrows, preview_offset)
+
+
+def _build_row_subset(
+    row: Union[List[Any], Tuple[Any]],
+    sh_name: str,
+    sheetnames: List[str],
+    row_number: int,
+    row_subset: List[str],
+) -> List[str]:
+    """
+    THis method will build each row and add an extra row for the sheet_name
+    If we're in an excel with multiple sheets
+
+    """
+
+    cells = [
+        str(cell.value).replace(",", " ")
+        if type(cell) not in [str, int, float, datetime.datetime] and cell is not None
+        else str(cell).replace(",", " ")
+        for cell in row
+    ]
+
+    if len(sheetnames) > 1:
+        row_subset.append(f'{",".join([*cells, sh_name])}\n')
+    else:
+        row_subset.append(f'{",".join(cells)}\n')
+
+    return row_subset
+
+
+def _get_row_subset_per_sheet(
+    wb: Union[openpyxl.workbook.Workbook, xlrd.book.Book],
+    sh_name: str,
+    sheetnames: List[str],
+    preview_nrows: Optional[int],
+    preview_offset: Optional[int],
+    row_subset: List[str],
+    skiprows: Optional[int] = None,
+    nrows: Optional[int] = None,
+    skipfooter: int = 0,
+) -> List[str]:
+    """
+    This method will get an iterator for from the workboot and
```

Comment:
Je ne comprends pas cette docstring ?  
Est ce que c'est ceci: 
```suggestion
    This method will get an iterator from the workbook and
```

Reply:
Thanks 👍🏾 
Updated !